src/Sylius/Behat/Context/Ui/Admin/ProductShowPageContext.php

```diff
@@ -334,4 +349,14 @@ public function iShouldSeeAttributeWithValueInLocale(string $attribute, string $
     {
         Assert::true($this->attributesElement->hasAttributeInLocale($attribute, $locale, $value));
     }
+
+    /**
+     * @Then /^I should see (this product) in ("([^"]*)" channel)$/
+     */
+    public function iShouldSeeThisProductInChannel(ProductInterface $product, ChannelInterface $channel): void
```

Comment:
I would be for moving this to a separate context and do not mix the admin and shop steps in one context